packages/app/search/service.ts

```diff
@@ -1,29 +1,54 @@
-import compile from 'monquery'
-import type { Model } from '../models/types'
+import type { HttpException } from 'utils/errors'
 import type { ErrorData } from '../declarations'
-import { searchModelsDb } from './db'
 import log from '../lib/log'
+import { getModel } from '../models/service'
+import { getSearchDb } from './db'
+import { searchInputServiceSchema } from './schema'
 
-/**this searvice takes lucene query string and converts it to mongoDB query*/
-
-export function searchwithMonquery(searchQuery) {
-    let search = compile(searchQuery)
-
-    return search
-}
-
-/*this function throws an error if there is a model users and workflows. since we do not want this models to be searched.*/
-export async function getModel(modelName: string): Promise<ErrorData<Model>> {
+export async function search(
+    modelName: string,
+    query: string,
+    resultsPerPage: number,
+    pageNumber: number
+): Promise<ErrorData<any>> {
     try {
-        const [modelError, model] = await getModel(modelName)
+        const searchDb = await getSearchDb()
+        const parsedInput = searchInputServiceSchema.parse({
+            modelName,
+            pageNumber,
+            query,
+            resultsPerPage,
+        })
+        const [modelError, { titleProperty }] = await getModel(modelName)
 
-        if (model.name.includes('Users' || 'Workflows')) {
+        if (modelError) {
             throw modelError
         }
 
-        return [null, model]
+        const searchResults = await searchDb.search(
+            parsedInput.modelName,
+            titleProperty,
+            parsedInput.query,
+            parsedInput.resultsPerPage,
+            parsedInput.pageNumber
+        )
+
+        return [null, searchResults]
     } catch (error) {
         log.error(error)
+
+        //* monquery throws an AssertionError for Lucene syntax errors.
+        if (error.name === 'AssertionError') {
+            return [formatAssertionError(error), null]
+        }
+
         return [error, null]
     }
 }
+
+//* This might be useful if directly integrated into the apiError function, but for now it's a one-off use case.
+function formatAssertionError(error: Error | HttpException) {
```

Comment:
Monquery has fairly descriptive error messages, but they don't seem to be the most accurate. E.g., monquery might complain about a missing ')' near "MERRA", but what it really needs is quotation marks around a search phrase that contains spaces. Still, this formatter helps give context and sometimes accurately describes the actual error.

Reply:
Thanks for adding this, it will be very helpful to determine actual error.